tempesta_fw/msg.c

```diff
@@ -62,24 +64,25 @@ tfw_msg_write(TfwMsgIter *it, const TfwStr *data)
 
 		memcpy_fast(p, c->data + c_off, n_copy);
 
-		if (c_size < f_room) {
-			/*
-			 * The chunk fits in the SKB fragment with room
-			 * to spare. Stay in the same SKB fragment, switch
-			 * to next chunk of the string.
-			 */
-			c_off = 0;
-		} else {
-			frag = ss_skb_frag_next(&it->skb, it->skb_head,
-						&it->frag);
+		/*
+		 * The chunk occupied all the spare space in the SKB fragment,
+		 * switch to the next fragment.
+		 */
+		if (c_size >= f_room) {
+			skb_frag_t *frag = ss_skb_frag_next(&it->skb,
+							    it->skb_head,
+							    &it->frag);
+			if (unlikely(frag)
```

Comment:
Doesn't look right.
I'd expect to return an error (-E2BIG), if there is no room in skb list and there are still data to be written. But if `frag != NULL`, there is room available, so we can just continue to fill the buffers.
Maybe, it should be `!frag` instead of `frag`?